lib/phoenix/live_dashboard/components/tab_bar_component.ex

```diff
@@ -0,0 +1,139 @@
+defmodule Phoenix.LiveDashboard.TabBarComponent do
+  @moduledoc false
+  use Phoenix.LiveDashboard.Web, :live_component
+
+  @impl true
+  def mount(socket) do
+    {:ok, socket}
+  end
+
+  @impl true
+  def update(assigns, socket) do
+    %{page: page, tabs: tabs} = assigns
+    current = current_tab(page.params, tabs)
+    {:ok, assign(socket, tabs: tabs, current: current, page: page)}
+  end
+
+  defp current_tab(params, tabs) do
+    with %{"tab" => tab} <- params,
+         true <- Enum.any?(tabs, fn {id, _} -> Atom.to_string(id) == tab end) do
+      String.to_existing_atom(tab)
+    else
+      _ -> default_tab(tabs)
+    end
+  end
+
+  defp default_tab([{id, _} | _]) do
+    id
+  end
+
+  def validate_params(params) do
+    with :ok <- validate_tabs(params),
+         do: validate_page(params)
+  end
+
+  defp validate_page(params) do
+    case Map.fetch(params, :page) do
+      :error ->
+        {:error, "expected :page parameter to be received"}
+
+      {:ok, %Phoenix.LiveDashboard.PageBuilder{}} ->
+        :ok
+
+      {:ok, invalid} ->
+        msg = "expected :page parameter to be a %PageBuilder{}, received: "
+        {:error, msg <> inspect(invalid)}
+    end
+  end
```

Comment:
We don't need to validate page because it is always set internally:

```suggestion
  def validate_params(params) do
    validate_tabs(params)
  end

```

:)